### PR TITLE
Allow for specifying multiple git URLs as fallbacks

### PR DESCRIPTION
**Pull request by no92**

### CODE REVIEW

File: xbstrap/vcs_utils.py [hunk 1]
```diff
@@ -223,7 +228,12 @@ def fetch_repo(cfg, src, subdir, *, ignore_mirror=False, bare_repo=False):
             else:
                 subprocess.check_call([git, "init"] + b_args, cwd=source_dir)
             # We always set the remote to the true remote, not a mirror.
-            subprocess.check_call([git, "remote", "add", "origin", source["git"]], cwd=source_dir)
+            subprocess.check_call([git, "remote", "add", "origin", git_sources[0]], cwd=source_dir)
+
+            for num, fallback in enumerate(git_sources[1:]):
```
Comment (ArsenArsen) [inline xbstrap/vcs_utils.py:234]: i'm not sure I like splitting this array like this - it hard-codes the assumption of one element in the mirror list.

do we even need to handle fallbacks specially?  why not just insert all sources as remotes and save the list of names for further use

Comment (no92) [reply xbstrap/vcs_utils.py:234]: We are guaranteed to have at least one element in the mirror list, as that's enforced by the schema. Having the primary remote be named `origin` is a convenience choice for the user that might use git inside the cloned package.

Comment (ArsenArsen) [reply xbstrap/vcs_utils.py:234]: hm, doesn't the schema accept `[]`?

Comment (no92) [reply xbstrap/vcs_utils.py:234]: Fair point. I amended the schema to require at least one list entry and added an error message if `[]` is passed.

**Review by ArsenArsen — APPROVED** [review]
seems ok, cc @avdgrinten

**Review by avdgrinten — APPROVED** [review]

**Pull request merged**